src/main/java/com/mathworks/ci/MatlabBuild.java

```diff
@@ -22,33 +22,29 @@
      * This Method decorates the launcher with MATLAB command provided and returns the Process
      * object to launch MATLAB with appropriate startup options like -r or -batch
      * 
+     * @param matlabLauncher current build launcher
      * @param workspace Current build workspace
      * @param launcher Current build launcher
      * @param listener Current build listener
-     * @param envVars Environment variables of the current build
      * @param matlabCommand MATLAB command to execute on shell
      * @return matlabLauncher returns the process launcher to run MATLAB commands
      */
-    default ProcStarter getProcessToRunMatlabCommand(FilePath workspace, Launcher launcher,
-            TaskListener listener, EnvVars envVars, String matlabCommand, String uniqueName)
+    default ProcStarter getProcessToRunMatlabCommand(ProcStarter matlabLauncher, FilePath workspace,
+            Launcher launcher, TaskListener listener, String matlabCommand, String uniqueName)
             throws IOException, InterruptedException {
         // Get node specific tmp directory to copy matlab runner script
         String tmpDir = getNodeSpecificTmpFolderPath(workspace);
         FilePath targetWorkspace = new FilePath(launcher.getChannel(), tmpDir);
-        ProcStarter matlabLauncher;
         if (launcher.isUnix()) {
             final String runnerScriptName = uniqueName + "/run_matlab_command.sh";
-            matlabLauncher = launcher.launch().pwd(workspace).envs(envVars)
-                    .cmds(tmpDir + "/" + runnerScriptName, matlabCommand).stdout(listener);
+            matlabLauncher.cmds(tmpDir + "/" + runnerScriptName, matlabCommand).stdout(listener);

```

Comment:
Why does the `runnerScriptName` start with a forward slash and then `tmpDir` is separated from it with a forward slash as well?

Reply:
That's because I create a tmpDir as UUID in keep it in system tmp directory with all helper files like runner script and now Matlab script and delete the entire folder at the end.